selfdrive/loggerd/logger.cc

```diff
@@ -12,12 +12,97 @@
 
 #include <pthread.h>
 #include <bzlib.h>
+#include <iostream>
+#include <fstream>
+#include <streambuf>
+#ifdef QCOM
+#include <cutils/properties.h>
+#endif
 #include "messaging.hpp"
 
 #include "common/swaglog.h"
-
+#include "common/params.h"
+#include "common/util.h"
+#include "common/version.h"
 #include "logger.h"
 
+void append_property(const char* key, const char* value, void *cookie) {
+  std::vector<std::pair<std::string, std::string> > *properties =
+    (std::vector<std::pair<std::string, std::string> > *)cookie;
+
+  properties->push_back(std::make_pair(std::string(key), std::string(value)));
+}
+
+kj::Array<capnp::word> gen_init_data() {
```

Comment:
let's make this log_gen_init_data to match the pattern we use for sentinel

Reply:
call `log_init_data` in `logger_init` now.Is this what you want

Reply:
yes